.travis.yml

```diff
@@ -8,8 +8,6 @@ addons:
     packages:
       - inkscape
       - kicad
-env:
-  - CACHED_BUILD=cached
```

Comment:
I think this is the source of the defect.

Reply:
What do you mean? We should be able to turn the env var on and off as we wish, including in the .yml file.

Reply:
If you look into the git history, you will find that I added it when I was trying to get the build to work with the cache without setting the environment variable directly on the Travis dashboard. We should be able to set the environment variables from the `.travis` file but honestly, I don't understand how Travis manages the build options.:sweat_smile:
Do you actively use the `.travis` file to set the environment variables? is it a must-have?

Reply:
I normally set it on Travis settings but Ian Ross uses the .travis.yml to set that var. It shouldn't make any difference how it's being set. One just overrides the other. You can see it being overriden e.g. here: https://travis-ci.org/github/kitspace/kitspace/builds/728930222#L545

Reply:
Here is a thing I didn't notice, even if this was the reason it shouldn't make the build fail when you run it locally. The fact that I can have the same build setup locally is annoying. Have you tried `git bisect`?

Reply:
I am not quite sure what you are saying.  Why is it annoying that you _can_ have the same build setup locally? Did you mean _can't_?

It would make sense to be consistent I guess: to only pass the env var in at one point or grab it from the env everywhere. But you can set the env var locally too using `export` or `CACHED_BUILD=false ./configure production false` etc. 

I've used `git bisect` before but not on this branch or kitspace at all yet.

Reply:
> I am not quite sure what you are saying. Why is it annoying that you _can_ have the same build setup locally? Did you mean _can't_?

Ah sorry for the typo!

> I've used `git bisect` before but not on this branch or kitspace at all yet.

I meant [a few days ago](https://github.com/kitspace/kitspace/pull/304#issuecomment-690748375) I asked you to run bisect of you have the time; to see when `boards.json` started not to have all the boards as you mentioned [here](https://github.com/kitspace/kitspace/pull/304#issuecomment-690733826).

The only hypothesis I could come up with is that the targets which need `boards.json` are already built and cached but in this case, there is no need to rebuild `boards.json` at all, not rebuild it as an empty array. Also, I haven't tried to test this hypothesis ye because not being able to configure a production configuration.
 
Also, I think it's actually good in this case to make the test fail and retrigger a fresh build without cache as it's invalid even though boards get rendered correctly to the homepage.

Reply:
This seems like a false positive.[729546081](https://travis-ci.org/github/kitspace/kitspace/builds/729546081) is failing while [729546085](https://travis-ci.org/github/kitspace/kitspace/builds/729546085) is passing. Both are built from the same commit the only difference is that they are different caches.

Reply:
Doesn't seem like a false positive to me. Seems like those two builds are showing exactly the problem we are trying to solve here.